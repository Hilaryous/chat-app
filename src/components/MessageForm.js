// @flow
import React, { PureComponent } from 'react'
import styled from 'react-emotion'
import AppContext from '../context'
import Input from './Input'
import Button from './Button'

type Props = {
  addMessageToList: ({}) => void,
  currentChatId: string,
  currentUser: {
    id: string,
    email: string,
  },
}

type State = {
  message: string,
}

class MessageForm extends PureComponent<Props, State> {
  initialState = { message: '' }

  state = this.initialState

  handleClearMessage = () => {
    this.setState(() => ({ message: '' }))
  };

  handleMessageChange = (e) => {
    this.setState({ message: e.target.value })
  };

  handleAddMessageToList = (e) => {
    e.preventDefault()
    const { message } = this.state
    const { addMessageToList, currentChatId, currentUser: { id } } = this.props
    const messageObj = {
      chatId: currentChatId,
      userId: id,
      content: message.trim(),
    }
    const cleanedMessage = message.trim()
    if (cleanedMessage.length >= 1) {
      addMessageToList(messageObj)
      this.handleClearMessage()
    }
  }

  render() {
    const { message } = this.state
    const { currentChatId } = this.props
    return (
      currentChatId != null && (
      <Form onSubmit={this.handleAddMessageToList}>
        <Input
          type="text"
          placeholder="Message"
          value={message}
          onChange={this.handleMessageChange}
        />
        <Button type="submit">
          send
        </Button>
      </Form>
      )
    )
  }
}

export default () => (
  <AppContext.Consumer>
    {({ currentChatId, currentUser, actions: { addMessageToList } }) => (
      <MessageForm
        addMessageToList={addMessageToList}
        currentUser={currentUser}
        currentChatId={currentChatId}
      />
    )
        }
  </AppContext.Consumer>
)

// ----------------------------------

const Form = styled('form')({
  width: '100%',
  display: 'grid',
  gridTemplateColumns: '3fr 1fr',
  gridGap: '16px',
  alignItems: 'end',
})
