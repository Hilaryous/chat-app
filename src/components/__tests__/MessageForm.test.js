import React from 'react'
import renderer from 'react-test-renderer'
import { StaticRouter } from 'react-router-dom'
import MessageForm from '../MessageForm'
import AppContext from '../../context'

describe('MessageForm', () => {
  it('shows the correct snapshot tree for the MessageForm component', () => {
    const tree = renderer.create(
      <StaticRouter location="someLocation" context={{}}>

        <AppContext.Provider>
          <MessageForm />
        </AppContext.Provider>
      </StaticRouter>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
