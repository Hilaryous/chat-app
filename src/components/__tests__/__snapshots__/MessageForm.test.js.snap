// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm shows the correct snapshot tree for the MessageForm component 1`] = `null`;
